controller/src/main/java/io/pravega/controller/server/rest/ModelHelper.java

```diff
@@ -156,4 +172,25 @@ public static final StreamProperty encodeStreamResponse(String scope, String str
         streamProperty.setRetentionPolicy(retentionConfig);
         return streamProperty;
     }
+
+    private static RetentionPolicy getRetentionPolicy(TimeBasedRetention timeRetention, long retentionInDays) {
+        Duration retentionDuration = (timeRetention != null && retentionInDays == 0) ?
+                Duration.ofDays(timeRetention.getDays())
+                        .plusHours(timeRetention.getHours())
+                        .plusMinutes(timeRetention.getMinutes())
+                :  Duration.ofDays(retentionInDays);
+        return RetentionPolicy.byTime(retentionDuration);
+    }
+
+    private static long getMinsFromMillis(long millis, long hours, long daysInMs) {
```

Comment:
the params for this method seem a bit odd, mills, hours and daysInMs.
can we make it consistent?

Reply:
I have added a javadoc and changed param names to make them more intuitive. Please check.